### PR TITLE
Add a warning for single ':'s in variable names

### PR DESCRIPTION
**Pull request by Nicofisi**
One way of implementing the idea described in #1440.

Leaving it here as a PR in case you think the warning could be worded better, or anything else

### CODE REVIEW

**Comment by Nicofisi** [comment]
Side-note: this works in scripts but doesn't in effect commands.. are they coded to ignore all warnings?

**Comment by TheBentoBox** [comment]
I personally think the error is a bit wordy, and it uses the word "separator" a lot since that's the name of the variable in the actual code, but to most end users the wording here doesn't make as much sense. Maybe something like `"':' should only be used in variable names to denote lists. Did you mean '::'?"`

You could also add a specific check for if the string contains `SINGLE_SEPARATOR_CHAR + '*'` to check very specifically for lists that accidentally have just one ':' since that's the primary case of this issue, as described in #1440 .

**Comment by Nicofisi** [comment]
> I personally think the error is a bit wordy, and it uses the word "separator" a lot since that's the name of the variable in the actual code, but to most end users the wording here doesn't make as much sense. 

I, ugh, "inspired" myself from this, as it uses the word separator too:
https://github.com/SkriptLang/Skript/blob/66563533e1dcae6c17903bed6107b1bce3a9eac7/src/main/java/ch/njol/skript/lang/Variable.java#L125
Do you think this should be changed to something more user friendly too?

> You could also add a specific check for if the string contains SINGLE_SEPARATOR_CHAR + '*' to check very specifically for lists that accidentally have just one ':' since that's the primary case of this issue, as described in #1440 .

Hmm, you mean instead, right? I'm not sure this is necessarily the best idea since then you could `set {something:%player%} to 123` and then `loop {something::*}:` and it wouldn't warn

**Comment by TheBentoBox** [comment]
I think at least the word "list" could be added to that other error ("... must neither start nor end with the **list** separator...").

Also for the other error suggested, I didn't mean instead, I just meant that we could give a more specific error message if we specifically detect a hard-coded ":*" in a list name. If that exists they're very obviously trying to write a list but accidentally only put one ':', in which case we can give a very specific message ("You need to use two list separators (::) to denote a list but you're only using one.") You could first check for ":*" to give a specific warning like that and if it's not there give the more general warning which dissuades from using singular ':' regardless of context.

Hope that makes sense, it's a bit hard to word stuff related to this 😂

File: src/main/java/ch/njol/skript/lang/Variable.java [hunk 1]
```diff
@@ -135,6 +136,11 @@ public static boolean isValidVariableName(String name, final boolean allowListVa
 			if (printErrors)
 				Skript.error("A variable's name must not contain the separator '" + SEPARATOR + "' multiple times in a row (error in variable {" + name + "})");
 			return false;
+		} else if (org.apache.commons.lang3.StringUtils.remove(name, SEPARATOR).contains(SINGLE_SEPARATOR_CHAR)) {
+			if (printErrors)
```
Comment (bensku) [inline src/main/java/ch/njol/skript/lang/Variable.java]: Maybe instead: If you meant to make this variable a list, it should contain two double colons (::); currently, it contains only one.

Comment (Pikachu920) [reply src/main/java/ch/njol/skript/lang/Variable.java]: that's getting there. i think it's still needs some refinement tho
